alerting/src/main/kotlin/org/opensearch/alerting/MonitorFanOutUtils.kt

```diff
@@ -0,0 +1,35 @@
+package org.opensearch.alerting
+
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.opensearch.Version
+import org.opensearch.cluster.node.DiscoveryNode
+
+private val logger: Logger = LogManager.getLogger("FanOutEligibility")
+
+fun isNodeEligibleForFanOut(
+    candidateNode: DiscoveryNode,
+    coordinatorNode: DiscoveryNode,
+): Boolean {
+    try {
+        val candidateNodeAttributes: Map<String, Any> = candidateNode.attributes
+        val coordinatorNodeAttributes: Map<String, Any> = coordinatorNode.attributes
+        if (candidateNodeAttributes.containsKey("di_number") && candidateNodeAttributes["di_number"] != null &&
+            coordinatorNodeAttributes.containsKey("di_number") && candidateNodeAttributes["di_number"] != null
+        ) {
+            return candidateNodeAttributes["di_number"] as Int >= coordinatorNodeAttributes["di_number"] as Int
```

Comment:
Is there a downside to keeping the fanout fully within the same DI (`==` instead of `>=`)? That seems like a safer approach than relying on backwards compatibility in newer DIs